var/spack/repos/builtin/packages/parsec/package.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+#
+from spack import *
+
+
+class Parsec(CMakePackage):
+    """PaRSEC: the Parallel Runtime Scheduler and Execution Controller
+
+    PaRSEC is a runtime and a programming toolbox that support the design and
+    parallel execution of micro-tasks on distributed, heterogeneous systems.
+    """
+
+    homepage    = "https://icl.utk.edu/dte"
+    git         = "https://bitbucket.org/icldistcomp/parsec.git"
+    url         = "https://bitbucket.org/icldistcomp/parsec/get/parsec-3.0.2012.tar.bz2"
+    list_url    = "https://bitbucket.org/icldistcomp/parsec/downloads/?tab=tags"
+    maintainers = ['abouteiller', 'bosilca', 'herault']
+
+    version('master', branch='master')
+    version('3.0.2012-rc1', sha256='a0f013bd5a2c44c61d3d76bab102e3ca3bab68ef2e89d7b5f544b9c1a6fde475')
+    version('1.1.0', sha256='d2928033c121000ae0a554f1e7f757c1f22274a8b74457ecd52744ae1f70b95a', url='https://bitbucket.org/icldistcomp/parsec/get/v1.1.0.tar.bz2')
+
+    variant('build_type', default='RelWithDebInfo', description='CMake build type', values=('Debug', 'Release', ' RelWithDebInfo'))
+    variant('shared', default=True, description='Build a shared library')
+    variant('cuda', default=True, description='Use CUDA for GPU acceleration')
```

Comment:
Did you consider inheriting also from `CudaPackage`:
```python
class Parsec(CMakePackage, CudaPackage):
```
? That would add automatically:
1. A few variants, like `cuda`
2. Constraints against compilers that are reported in the CUDA documentation

Reply:
I am looking into it, how can I set `+cuda` to be the default?

Reply:
Just add:
```python
variant('cuda', default=True, description='Build with CUDA')
```
to override the variant default.